nucleus/common/common-util/src/main/java/com/sun/enterprise/security/store/AsadminSecurityUtil.java

```diff
@@ -252,7 +253,30 @@ private void init(final char[] commandLineMasterPassword, final boolean isPrompt
          * The keystore has been opened successfully, using passwordToUse.
          * Open the truststore with that password.
          */
-        asadminTruststore = openTruststore(passwordToUse);
+        try {
+            asadminTruststore = openTruststore(passwordToUse);
+        } catch (IOException e) {
+            logger.log(Level.WARNING, String.format("Error when reading truststore, exception:%s. Now recreating file", e));
+            recreateTrustStore(AsadminTruststore.getAsadminTruststore(), passwordToUse);
+        }
+    }
+
+    /**
+     * Method to recreate the truststore file
+     * @param fileTruststore File of the truststore
+     * @param passwordToUse password to use
+     * @throws IOException
+     */
+    private void recreateTrustStore(File fileTruststore, char[] passwordToUse) throws IOException {
+        try {
+            Files.deleteIfExists(fileTruststore.toPath());
```

Comment:
This might be a bit blanket.
A user can define a custom truststore for asadmin using the standard java system property, so as-is you'd potentially be deleting someone's custom truststore.

See the below method from `AsadminTruststore` for inspiration on determining if `fileTruststore` is the default generated one or not:

```
    public static File getAsadminTruststore()
    {
        String location = System.getProperty(SystemPropertyConstants.CLIENT_TRUSTSTORE_PROPERTY);
        if (location == null) {
            return new File(AsadminSecurityUtil.getDefaultClientDir(), ASADMIN_TRUSTSTORE);
        } else {
            return new File(location);
        }
    }
```

Reply:
This trust store is not the one that the user defines. It just contains the certificate (if you accept) of the server when performing asadmin commands in secure mode.

The trust store never contains any user defined artifacts (as it is the one in the .gfclient directory)

Reply:
I already added a validation to recreate the default truststore from the gfclient folder in case the SystemPropertyConstants.CLIENT_TRUSTSTORE_PROPERTY is not set

Reply:
> This trust store is not the one that the user defines. It just contains the certificate (if you accept) of the server when performing asadmin commands in secure mode.
> 
> The trust store never contains any user defined artifacts (as it is the one in the .gfclient directory)

My argument is that it's _not necessarily_ the one in the gfclient folder. The option is present for you to define a custom one. See the snippet I posted originally - it checks if `-Djavax.net.ssl.trustStore` is set and uses that if defined.

The ~/.gfclient is also technically customisable via 
```
                String AS_GFCLIENT = System.getenv("AS_GFCLIENT");
		logger.log(Level.FINER, "AS_GFCLIENT: {0}", AS_GFCLIENT);
		if (AS_GFCLIENT != null) {
    			DEFAULT_CLIENT_DIR = new File(AS_GFCLIENT);
		} else {
    			DEFAULT_CLIENT_DIR = new File(System.getProperty("user.home"), ".gfclient");
		}
```